extensions/amp-base-carousel/0.1/test-e2e/test-rtl.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  getNextArrow,
+  getPrevArrow,
+  getSlide,
+} from './helpers';
+
+const pageWidth = 800;
+const pageHeight = 600;
+
+describes.endtoend('AMP carousel', {
+  testUrl: 'http://localhost:8000/test/manual/amp-base-carousel/basic-rtl.amp.html',
+  experiments: ['amp-base-carousel', 'layers'],
+  initialRect: {width: pageWidth, height: pageHeight},
+  // TODO(sparhami) Make other environments work too
+  environments: ['single'],
```

Comment:
When the tests fail in the viewer-demo and shadow-demo environments, are you able to verify that the behaviors work correctly when you manually load the document in the environments? e.g. http://localhost:8000/examples/viewer.html#href=http://localhost:8000/test/manual/amp-base-carousel/basic-rtl.amp.html and http://localhost:8000/pwa#href=http://localhost:8000/test/manual/amp-base-carousel/basic-rtl.amp.html

Reply:
They do work locally. I just need to spend some time digging into why it isn't working. I think it is more the window size / test page size than anything.